thunder/benchmarks/__init__.py

```diff
@@ -2958,3 +2960,163 @@ def fn(self) -> Callable:
+    else:
+        raise TypeError(
+            f"The 'example_value' in the placeholder node is expected to be either a Tensor or a Tuple of Tensors."
+        )
+
+
+class ThunderCompilerGraphBenchmarking(ThunderCompiler):
+    _executors = (
+        "eager",
+        "inductor",
+        "thunder",
+    )
+
+    def __init__(
+        self,
+        bench: BenchmarkFixture,
+        executors: Sequence[str] | None = None,
+        **thunder_options,
+    ):
+        """
+        This class acts as a backend for the `torch.compile` function, facilitating the benchmarking of each `fx.GraphModule` produced by Thunder dynamo splitter.
+        Each `fx.GraphModule` instance is executed by the specified executors and benchmarked using `pytest_benchmark`.
+
+        Keyword arguments:
+            bench: the BenchmarkFixture created by `pytest_benchmark`
+            executors: List of executors to use. Supported executors include: 'eager', 'inductor', and 'thunder'. If None, defaults to all available executors.
+            thunder_options: a dictionary of options to pass to `thunder.jit`. Besides all the arguments to `thunder.jit`,
+                             it accepts `torch_inductor_options` which are passed to `torch.compile` if part of the graph
+                             is not supported by thunder.
+
+        Example:
+            >>> import torch
+            >>> import thunder
+            >>> from thunder.benchmarks import ThunderCompilerGraphBenchmarking
+            >>>
+            >>> def func(x):
+            ...     x = torch.sin(x)
+            ...     if x.sum() > 0:
+            ...         return x + 1
+            ...     else:
+            ...         return x - 1
+            ...
+            >>> def test_func(benchmark):
+            ...     backend = ThunderCompilerGraphBenchmarking(benchmark, executors=["eager"])
+            ...     compiled = torch.compile(backend=backend)(func)
+            ...     x = torch.ones(2, requires_grad=True).cuda()
+            ...     compiled(x)
+        Running the example with `pytest script.py  --benchmark-sort="name"` will produce benchmarking results.
+
+        # Dynamo segments the graph into two subgraphs, each identified by the 'GraphId[id]' field in the test name.
+        # Each subgraph contains a single split module, processed by the Thunder-defined splitter,
+        # which corresponds to the 'SplitModuleName[split_module_name]' field.
+        # The currently active executor is indicated by the '_eager' suffix.
+        ----------------------------------------------------------------------------------------------------- benchmark: 2 tests -----------------------------------------------------------------------------------------------------
+        Name (time in us)                                             Min                Max               Mean            StdDev             Median               IQR            Outliers  OPS (Kops/s)            Rounds  Iterations
```

Comment:
Looking at this example, I'm speculating that the memory foot print is not displayed even though we seem to record peak memory consumption.
Would it be possible to have that as well in the table or in a separate one?

Reply:
From what I've investigated, the content of the table that is printed in pytest-benchmark seems to be fixed, the extensible interface is `extra_info` which is only shown in the json file. So I put the memory info there

Reply:
Right, unfortunately, extra information can be printed only with some other custom tools. That was also the reason RAPIDS project [has forked pytest-benchmark](https://github.com/rapidsai/benchmark).